Gems/ONNX/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+set(gem_path ${CMAKE_CURRENT_LIST_DIR})
+set(gem_json ${gem_path}/gem.json)
+
+add_compile_definitions("ENABLE_CUDA=true") # Change to ENABLE_CUDA=false to disable CUDA inferencing
+
+add_compile_definitions("W_GEM_ASSETS_PATH=L\"${CMAKE_CURRENT_LIST_DIR}/Assets\"" "GEM_ASSETS_PATH=\"${CMAKE_CURRENT_LIST_DIR}/Assets\"")
```

Comment:
add_compile_definitions adds these compile definitions to the entire build solution, which is not allowed.
These compiler definitions should be tied to the ONNX `ly_add_target` command  in the `Gems/ONNX/Code/CMakeLists.txt` via the COMPILE_DEFINITIONS argument
https://github.com/o3de/o3de/blob/86343d17f4e3c154c7dc1c32e4c9ef5d2173c717/cmake/LYWrappers.cmake#L85

Reply:
Also these defines to retrieve the Gem Asset folder is not portable and should be removed.
I will provide instructions on how to portability access the gem directory in future comments.

Reply:
Changed